src/cli/cluster/base_path_proxy.js

```diff
@@ -53,6 +53,28 @@ export default class BasePathProxy {
     server.route({
       method: '*',
       path: `${basePath}/{kbnPath*}`,
+      config: {
+        pre: [
+          (req, reply) => {
+            map(clusterManager.workers, (worker) => {
+              if (worker.type === 'server' && !worker.listening && !worker.crashed) {
+                return fromNode(cb => {
+                  const done = () => {
```

Comment:
Well, wouldn't keeping the events attached to the worker cause a memory leak eventually?